include/core/truthTable/truth_table.hpp

```diff
@@ -112,6 +133,16 @@ namespace syrec {
                 cube.pop_back();
             }
 
+            [[nodiscard]] auto equals(const std::uint64_t num) const -> bool {
+                Cube cubeEq(cube);
+                return cubeEq.toInteger() == num;
+            }
```

Comment:
Avoid creating unnecessary copies of objects.
```c++
            [[nodiscard]] auto equals(const std::uint64_t num) const -> bool {
                return cube.toInteger() == num;
            }
```
But, even more important, the two functions here use different kinds of logic for doing the comparison. This should much rather be (which also properly works If the stored cube has more than 64bits)
```suggestion
            [[nodiscard]] auto equals(const std::uint64_t num) const -> bool {
                return cube == Cube::fromInteger(num);
            }
```

Reply:
is there a way to get the length of occupied bits from the uint32_t ? I don't think I can overload the equals function then.

Reply:
You could just add an initial "bw" bitwidth parameter just as for the other methods taking an integer.

Reply:
done 👍🏼